src/rules/valid-expect.ts

```diff
@@ -249,20 +249,15 @@ export default createRule<
 
         const parentNode = matcher.node.parent;
 
-        if (
-          !modifier ||
-          !parentNode ||
-          !parentNode.parent ||
-          modifier.name === ModifierName.not
-        ) {
+        if (!modifier || modifier.name === ModifierName.not) {
```

Comment:
We don't wanna bail early?

Reply:
Do you mean for `parentNode.parent`?

It's actually an impossibility: the only ast node that can not have a parent is `Program`.

See https://github.com/typescript-eslint/typescript-eslint/issues/1417

Reply:
We have an optional chain further down, tho. I'd prefer an early return or a type refinement which captures the impossibility

Reply:
Fair enough 🙂